tests/rptest/tests/topic_delete_test.py

```diff
@@ -196,13 +197,13 @@ def _populate_topic(self, topic_name):
         self.kafka_tools.alter_topic_config(
             topic_name, {'retention.local.target.bytes': 5 * 1024 * 1024})
 
-        # Write out 10MB
+        # Write out 10MB per partition
         self.kafka_tools.produce(topic_name,
                                  record_size=4096,
-                                 num_records=2560)
+                                 num_records=2560 * self.partition_count)
 
         # Wait for segments evicted from local storage
-        for i in range(0, 3):
+        for i in range(0, self.partition_count):
             wait_for_segments_removal(self.redpanda, topic_name, i, 5)
```

Comment:
Maybe `wait_for_segments_removal` should fail if the initial number of segments is less than the final desired count.